### PR TITLE
[Microsoft] Migrate parents in core to correct order

### LINKED ISSUE #6975: [Vaults] Migrate parents in core to correct order for microsoft
Parents should be : first document Id, then parent id, then grandparent id etc.
For microsoft, they are in inverse order

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
Fixes #6975

Remove the reverse making parents in the wrong order, & add the migration script

Risks
---
Radius: microsoft connectors
Risk: messing up with exisiting accounts, but there are only 4; worst case we can full resync

Deploy plan
---
- pause all microsoft connectors
- deploy connectors
- run migration
- restart connectors

### CODE REVIEW

File: connectors/migrations/20240828_microsoft_refill_parents_field.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        connectorId: connector.id,
+        internalId: node.internalId,
+        startSyncTs,
+      });
+
+      if (!execute) {
+        return true;
+      }
+
+      return updateDocumentParentsField({
```
Comment (tdraier) [inline connectors/migrations/20240828_microsoft_refill_parents_field.ts:53]: What happens if the node has not been upserted as a document ? (i.e. if it was. a csv, or for another reason)

**Review by flvndvd — APPROVED** [review]
Thank you 🙏

File: connectors/migrations/20240828_microsoft_refill_parents_field.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    await updateParentsFieldForConnector(connector, execute);
+  }
+});
+
+async function updateParentsFieldForConnector(
+  connector: ConnectorResource,
+  execute = false
+) {
+  // get all nodes (not using resource here since 1-shot clean migration)
+  const nodes = await MicrosoftNodeModel.findAll({
```
Comment (tdraier) [inline connectors/migrations/20240828_microsoft_refill_parents_field.ts:24]: We need to also get tables, and update the parents list with `updateTableParentsField`

Comment (tdraier) [reply connectors/migrations/20240828_microsoft_refill_parents_field.ts:24]: ok looks we don't have the issue with tables, no issue here

**Review by tdraier — APPROVED** [review]

**Pull request merged**